refactor/ui-packaging

BassantM opened the pull request:
# What does this PR do?
<!-- Brief description of changes -->

## Type of Change
- [x] 🔧 Refactoring
- [ ] 🎨 UI changes

## Changes Made
- [x] Compose UI

## Checklist
- [x] Ready for review

BassantM committed:
refactor: update package structure in ui

BassantM committed:
refactor:
- replace content in screen to be private
- replace constants by companion objects
- rename some functions
- make some functions public to help for testing

Commit:
Feature/remove unused dependencies in gradle (#581)

BassantM committed:
refactor: account info package

BassantM committed:
refactor: home package in ui

ahmeddsamii committed:
feat: Add unit tests for TvShowDetailsViewModel (#582)

* feat: Add unit tests for TvShowDetailsViewModel

* Refactor: Make data initialization private in TvShowDetailsViewModel

* refactor: Remove firebase-crashlytics-ktx dependency

BassantM committed:
refactor: update ui packaging and component structure in search

BassantM committed:
Refactor: update UI packaging and component structure

marwanqashwa commented on presentation/src/main/java/com/london/presentation/feature/accountinfo/appearance/AppearanceBottomSheet.kt:
```diff
@@ -141,7 +141,7 @@ fun AppearanceBottomSheetContent(
 @ThemePreviews
 @Composable
 fun AppearanceBottomSheetContentPreview() {
```
Make it private

marwanqashwa commented on presentation/src/main/java/com/london/presentation/feature/accountinfo/language/LanguageBottomSheet.kt:
```diff
@@ -138,7 +138,7 @@ fun LanguageBottomSheetContent(
 @ThemePreviews
 @Composable
 fun LanguageBottomSheetContentPreview() {
```
Same

marwanqashwa commented on presentation/src/main/java/com/london/presentation/feature/accountinfo/logout/logoutBottomSheet.kt:
```diff
@@ -120,7 +120,7 @@ fun LogoutBottomSheetContent(
 @Composable
 fun LogoutBottomSheetContentPreview() {
```
Same

BassantM committed:
make preview functions private in appearance and logout

karimdeabes commented on presentation/src/main/java/com/london/presentation/feature/home/toprated/TopRatedScreen.kt:
```diff
@@ -41,7 +41,7 @@ import com.london.presentation.shared.buildscreen.BuildScreen
+import com.london.presentation.utils.gridColumns
 import com.london.presentation.utils.isLoading
 
 @Composable
```
i think you forgot to reordering the parameters here

karimdeabes commented on presentation/src/main/java/com/london/presentation/feature/home/trending/actor/TrendingActorsViewModel.kt:
```diff
@@ -16,6 +16,12 @@ class TrendingActorsViewModel @Inject constructor(
 
+    override fun onActorClick(id: Int) = emitEffect(TrendingActorsEffect.NavigateToActor(id))
+
+    override fun onBack() = emitEffect(TrendingActorsEffect.NavigateBack)
```
split the methods to separate lines

karimdeabes commented on presentation/src/main/java/com/london/presentation/feature/home/trending/movie/TrendingMoviesViewModel.kt:
```diff
@@ -17,6 +17,22 @@ class TrendingMoviesViewModel @Inject constructor(
+        initializeMovies()
+    }
+
+    override fun onBack() = emitEffect(TrendingMoviesEffect.NavigateBack)
```
also here separation make it more readable

karimdeabes commented on presentation/src/main/java/com/london/presentation/feature/home/trending/tvshow/TrendingTvShowsViewModel.kt:
```diff
@@ -18,6 +18,19 @@ class TrendingTvShowsViewModel @Inject constructor(
+        initializeTvShows()
+    }
+
+    override fun onTvShowClick(id: Int) = emitEffect(TrendingTvShowsEffect.NavigateToTvShow(id))
```
just the long line separate it

karimdeabes commented on presentation/src/main/java/com/london/presentation/feature/home/HomeScreen.kt:
```diff
@@ -69,7 +69,7 @@ import com.london.presentation.shared.GenresSection
+import com.london.presentation.utils.gridColumns
 
 @Composable
 fun HomeScreen(
```
this parameters not follow hierarchical approach

karimdeabes commented on presentation/src/main/java/com/london/presentation/feature/home/HomeViewModel.kt:
```diff
@@ -50,6 +50,76 @@ class HomeViewModel @Inject constructor(
+    }
+
+    override fun onMovieClick(id: Int) {
+        emitEffect(HomeScreenEffect.NavigationMovieDetails(id))
```
use compact function

karimdeabes commented on presentation/src/main/java/com/london/presentation/feature/home/HomeViewModel.kt:
```diff
@@ -50,6 +50,76 @@ class HomeViewModel @Inject constructor(
+        loadUpcomingMovies(categoryId = if (genre == MovieGenre.All) null else genre.id)
+    }
+
+    override fun onTopRatedClick() {
```
also here compact fun and all emit effect functions in the file

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/accountinfo/account/AccountScreen.kt:
```diff
@@ -29,19 +27,19 @@ import com.london.presentation.utils.Listen
     onNavigateToMyRating: () -> Unit = {},
-    onNavigateToLogin: () -> Unit = {}
+    onNavigateToWatchingHistory: () -> Unit = {},
+    viewModel: AccountViewModel = hiltViewModel()
```
Add comma after the last parameter

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/search/SearchContract.kt:
```diff
@@ -5,19 +5,19 @@ import com.london.domain.entity.recent.RecentSearch
+    fun removeRecentSearch(search: RecentSearch)
+    fun addToRecentSearches(query: RecentSearch)
+    fun onSearchQueryChange(newValue: TextFieldValue)
+    fun onCategorySelected(category: SearchCategory)
```
this line break the pyramid shape

Abdulrahman-Ragab-01 commented on presentation/src/main/java/com/london/presentation/feature/search/details/actor/ActorDetailsUiState.kt:
```diff
@@ -7,23 +7,23 @@ import com.london.domain.entity.actordetails.actortvshow.ActorTvShowDetails
-    val expanded: Boolean = false
+    val actorDetails: ActorDetails = ActorDetails(),
+    val actorImageDetails: List<ImageDetails>? = null,
+    val actorMovieDetails: ActorMovieDetails? = null,
```
shift this line above line 26 to make a beautiful pyramid

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/accountinfo/account/AccountViewModel.kt:
`initializeAppLanguage`, `observeContentRestrictionLevel`, `initializeAppTheme` move private functions to be after the overrides.

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/authentication/register/RegistrationScreen.kt:
```diff
@@ -0,0 +1,82 @@
+            is RegistrationEffect.RegistrationComplete -> onRegistrationComplete()
+        }
+    }
+    Content(
```
Add an empty line before this call.

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/authentication/register/RegistrationScreen.kt:
```diff
@@ -0,0 +1,82 @@
+        state = uiState,
+        contract = viewModel
+    )
+
```
Remove this empty line.

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/home/toprated/TopRatedUiState.kt:
```diff
@@ -10,14 +10,15 @@ import kotlinx.coroutines.flow.Flow
+    val selectedMediaCategory: MediaCategory = MediaCategory.MOVIES
 )
 
 
+
```
Remove empty lines.

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/home/toprated/TopRatedUiState.kt:
```diff
@@ -10,14 +10,15 @@ import kotlinx.coroutines.flow.Flow
+    val selectedMediaCategory: MediaCategory = MediaCategory.MOVIES
 )
 
 
```
Remove extra empty lines, it should only be one line at the end.

BassantM committed:
refactor: rearrange home view model, top rated screen, trending tv show view model

BassantM committed:
refactor: update HomeScreen navigation params order

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/search/SearchEffect.kt:
```diff
@@ -1,7 +1,7 @@
 package com.london.presentation.feature.search
 
 sealed interface SearchEffect {
+    data class TvNavigation(val tvId: Int) : SearchEffect
```
Please change name to something more descriptive.

BassantM committed:
refactor: reorder actor image details in state

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/search/SearchContract.kt:
```diff
@@ -5,19 +5,19 @@ import com.london.domain.entity.recent.RecentSearch
+    fun onClickMovie(genresListId : List<Int>)
+    fun removeRecentSearch(search: RecentSearch)
+    fun addToRecentSearches(query: RecentSearch)
+    fun onSearchQueryChange(newValue: TextFieldValue)
```
Move this below.

BassantM committed:
refactor: add missing onSearchQueryChange in SearchContract

BassantM committed:
style: fix line breaks in ViewModel functions

BassantM committed:
fix: rename TvNavigation to TvShowNavigation and remove extra empty lines

yusufnasserdev commented on presentation/src/main/java/com/london/presentation/feature/search/SearchViewModel.kt:
Please recheck the functions order, public should be first.

BassantM committed:
refactor: make functions as a compact functions

BassantM committed:
refactor: make public functions first

yusufnasserdev committed:
Merge branch 'develop' into refactor/ui-packaging

# Conflicts:
#	presentation/src/main/java/com/london/presentation/feature/search/details/tvshow/info/TvShowDetailsViewModel.kt

yusufnasserdev committed:
fix: Update My Lists screen title

- Changed "My List Screen" to "My Lists" in English strings.
- Updated Arabic translation for "My Lists".
- Ensured consistent title usage in the bookmarksScreen.
